### PR TITLE
Added ability to download the Qr Code by a button.

### LINKED ISSUE #5454: Add Ability to Download QR Code of Profile Link for Social Media Sharing
### Description

Linkfree allows users to create a profile with their links and share it with others. However, it would be useful if users could also download a QR code of their profile link that they could share on social media platforms. This feature would enable users to promote their links and profile more effectively, making it easier for others to access them.

Adding the ability to download a QR code of a Linkfree profile link would offer several advantages to users:
- Users would be able to share their links on social media platforms easily
- It would enable them to promote their Linkfree profile offline, for example, on business cards and flyers.
- Providing a QR code for their profile link would make it more convenient for others to access their links without needing to type the URL manually.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by MAYANKpandey14**
## Fixes Issue

closes #5454

## Changes proposed
Ability to download the Qr code of user's profile , which is accomplished by adding a Button. 
Button has to be shown only when user clicks on the Qr badge over the user's profile.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

#### Desktop :-

Screenshot 2023-03-16 at 10 13 34 PM

#### Mobile :-
Screenshot 2023-03-16 at 10 14 06 PM

## Note to reviewers
I have installed a npm package - filesaver.js , which helps in downloading the Qr code in PNG form, this package is handled by a function named downloadQR which is triggered onclick of the button.
Also changed the Qr creation method from QRCodeSVG to QRCodeCanvas so that the above mentioned function can convert it to png and become ready to download.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5539"></a>

### CODE REVIEW

File: components/user/UserProfile.js
Comment (eddiejaoude) [inline components/user/UserProfile.js:23]: Both functions should be consistent and use es6 style

```suggestion
  const downloadQR = () => {
    const qrCode = document.getElementById("qrcode");
    qrCode.toBlob((blob) => saveAs(blob, `linkfree-${data.username}.png`));
  };
```

File: components/user/UserProfile.js
Comment (eddiejaoude) [inline components/user/UserProfile.js:89]: Formatting is not consistent with the formatting changes on `71`

```suggestion
      <div className="flex justify-center mb-4">
        {qrShow && (
          <Button text="Download QR code" primary={true} onClick={downloadQR} />
        )}
      </div>
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks great 👍 

Just a couple of small inline changes

File: components/user/UserProfile.js
Comment (MAYANKpandey14) [reply components/user/UserProfile.js:23]: Thanks @eddiejaoude for the feedback , username in the file name will look cool .

**Comment by MAYANKpandey14** [comment]
> Looks great 👍
> 
> Just a couple of small inline changes

Thanks @eddiejaoude . I have committed the requested changes , your idea to add username in the file name was good 👍
Well this marks my 10th pull request.
Great community 💯, Thank you guys for helping me ❤️

**Comment by MAYANKpandey14** [comment]
Hey @eddiejaoude , there is a conflict due to the npm package i have added named file-saver , only user with write access can resolve this .
Kindly look into this once you're free.

File: components/user/UserProfile.js
Comment (CBID2) [inline components/user/UserProfile.js:89]:
```suggestion
      <article className="flex justify-center mb-4">
        {qrShow && (
          <Button text="Download QR code" primary={true} onClick={downloadQR} />
        )}
      </article>
```

Comment (CBID2) [inline components/user/UserProfile.js:89]:
```suggestion
      <article className="flex justify-center mb-4">
        {qrShow && (
          <Button text="Download QR code" primary={true} onClick={downloadQR} />
        )}
      </article>
```

File: components/user/UserProfile.js
Comment (CBID2) [inline components/user/UserProfile.js:25]:
```suggestion
      <article className="flex justify-center items-center flex-col md:flex-row gap-x-6">
```

File: components/user/UserProfile.js
Comment (CBID2) [inline components/user/UserProfile.js:76]:
```suggestion
      <article className="flex justify-center my-4">
```

File: components/user/UserProfile.js
Comment (CBID2) [inline components/user/UserProfile.js:84]:
```suggestion
      </article>
```

File: components/user/UserProfile.js
Comment (CBID2) [inline components/user/UserProfile.js:89]:
```suggestion
      <article className="flex justify-center mb-4">
        {qrShow && (
          <Button text="Download QR code" primary={true} onClick={downloadQR} />
        )}
      </article>
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @MAYANKpandey14! :) I replaced the `<div>`tags with `<article>` tags. It's always best to use semantic HTML tags because they help make lines of code more accessible. Other than that, great job! 👍🏾

File: components/user/UserProfile.js
Comment (eddiejaoude) [reply components/user/UserProfile.js:76]: I don't think using the `article` tag here is correct 

> The <article> tag specifies independent, self-contained content.
> An article should make sense on its own and it should be possible to distribute it independently from the rest of the site.
> Potential sources for the <article> element:
> - Forum post
> - Blog post
> - News story

https://www.w3schools.com/tags/tag_article.asp

Comment (MAYANKpandey14) [reply components/user/UserProfile.js:76]: > I don't think using the `article` tag here is correct
> 
> > The  tag specifies independent, self-contained content.
> > An article should make sense on its own and it should be possible to distribute it independently from the rest of the site.
> > Potential sources for the  element:
> > 
> > * Forum post
> > * Blog post
> > * News story
> 
> https://www.w3schools.com/tags/tag_article.asp

@CBID2 I agree with Eddie, div is more than enough for this specific task

**Comment by eddiejaoude** [comment]
> Hey @eddiejaoude , there is a conflict due to the npm package i have added named file-saver , only user with write access can resolve this .
> Kindly look into this once you're free.

In your PR you can resole the conflict. But I can do it if it helps

**Comment by MAYANKpandey14** [comment]
> > Hey @eddiejaoude , there is a conflict due to the npm package i have added named file-saver , only user with write access can resolve this .
> > Kindly look into this once you're free.
> 
> In your PR you can resole the conflict. But I can do it if it helps

This is what I am being shown . 
71C12BB2-1FCD-4D29-9DDF-B7B762FFC134

**Comment by eddiejaoude** [comment]
Yes only maintainers can merge it, but you can make any changes in your PR. 

I suggest running `npm install` and pushing the changes, then I can fix the rest

**Comment by MAYANKpandey14** [comment]
> Yes only maintainers can merge it, but you can make any changes in your PR.
> 
> I suggest running `npm install` and pushing the changes, then I can fix the rest

Okay ✅ thanks Eddie.

**Review by eddiejaoude — APPROVED** [review]
I will try to resolve conflicts via a temporary branch so we can get it merged

**Comment by MAYANKpandey14** [comment]
> I will try to resolve conflicts via a temporary branch so we can get it merged

Sure Eddie . Tag me when you need me I’ll be happy to help

**Comment by MAYANKpandey14** [comment]
> I will try to resolve conflicts via a temporary branch so we can get it merged

Thank you Eddie for understanding ❤️

**Comment by eddiejaoude** [comment]
The challenge is I can't merge it with conflicts and I can't make changes to your fork, so I am trying to work on a solution

The lock file is the biggest issue, it is easy to fix once I get it merged, if you generated a new lock file it would solve the problem but I am trying to find a work around

**Comment by MAYANKpandey14** [comment]
> The challenge is I can't merge it with conflicts and I can't make changes to your fork, so I am trying to work on a solution
> 
> The lock file is the biggest issue, it is easy to fix once I get it merged, if you generated a new lock file it would solve the problem but I am trying to find a work around

Give me few hours . I’ll do it don’t worry , I have asked someone to bring my laptop .

**Comment by eddiejaoude** [comment]
Please don't worry, nothing urgent. I think I have found a work around - it was a good learning for me as I am sure I will need to do it again for other PRs

I created a PR here if you want to have a look https://github.com/EddieHubCommunity/LinkFree/pull/5736

**Comment by MAYANKpandey14** [comment]
> Please don't worry, nothing urgent. I think I have found a work around - it was a good learning for me as I am sure I will need to do it again for other PRs
> 
> I created a PR here if you want to have a look #5736

Thank you @eddiejaoude . For understanding my situation and going out of the way to help me .

**Pull request merged**
